src/components/SearchView.tsx

```diff
@@ -0,0 +1,81 @@
+import * as React from "react";
+import { LibraryItem } from "./LibraryItem";
+import { LibraryView } from "../LibraryView";
+import { search, ItemData } from "../LibraryUtilities";
+
+interface SearchViewProps {
+    onSearchChanged: any;
+    libraryView: LibraryView;
+    items: ItemData[];
+}
+
+interface SearchViewStates {
+    searchText: string;
+}
+
+export class SearchView extends React.Component<SearchViewProps, SearchViewStates> {
+
+    constructor(props: SearchViewProps) {
+        super(props);
+
+        this.state = {
+            searchText: ''
+        };
+    }
+
+    performSearch(text: string) {
+        this.props.items.forEach(item => search(text, item));
+    }
```

Comment:
Move `performSearch` to utility class and add tests for it, maybe name it `searchItemRecursive`.